doc/gnoi-reboot/gnoi_reboot.md

```diff
@@ -0,0 +1,252 @@
+
+# **gNOI Reboot**
+
+
+## Table of Content
+
+
+<!-- TOC start (generated with https://github.com/derlin/bitdowntoc) -->
+
+- [Revision](#revision)
+- [Scope](#scope)
+- [Definitions/Abbreviations](#definitionsabbreviations)
+- [Overview](#overview)
+- [Requirements](#requirements)
+- [Architecture Design](#architecture-design)
+   * [Framework Container](#framework-container)
+- [High-Level Design](#high-level-design)
+   * [Telemetry &lt;-> Reboot BE API](#telemetry-lt-reboot-be-api)
+   * [Reboot Request Notification Schema](#reboot-request-notification-schema)
+   * [Reboot Response Notification Schema](#reboot-response-notification-schema)
+   * [CLI](#cli)
+   * [Sonic Host Services](#sonic-host-services)
+   * [SAI API {#sai-api}](#sai-api-sai-api)
+   * [Configuration and management {#configuration-and-management}](#configuration-and-management-configuration-and-management)
+   * [Warmboot and Fastboot Design Impact {#warmboot-and-fastboot-design-impact}](#warmboot-and-fastboot-design-impact-warmboot-and-fastboot-design-impact)
+   * [Restrictions/Limitations {#restrictions-limitations}](#restrictionslimitations-restrictions-limitations)
+   * [Testing Requirements/Design {#testing-requirements-design}](#testing-requirementsdesign-testing-requirements-design)
+
+<!-- TOC end -->
+
+
+
+<!-- TOC --><a name="revision"></a>
+
+
+
+## Revision
+
+
+Rev  | RevDate    | Author(s)   | Change Description
+---- | ---------- | ----------- | ------------------
+v0.1 | 09/22/2023 | John Hamrick (Google) | Initial Version
+
+
+
+## Scope
+
+_This document describes the high level design for SONiC gNOI Reboot RPC support._
+
+
+## Definitions/Abbreviations
+
+API: Application Programming Interface
+
+BE: Backend
+
+FE: Frontend
+
+gNOI: gRPC Network Operations Interface
+
+RPC: remote procedure call
+
+SAI: Switch Abstraction Interface
+
+SONiC: Software for Open Networking in the Cloud
+
+UMF: Unified Management Framework
+
+
+## Overview
+
+_Openconfig system.proto ([https://github.com/openconfig/gnoi/blob/main/system/system.proto](https://github.com/openconfig/gnoi/blob/main/system/system.proto)) defines reboot, cancel reboot and reboot status rpc’s within the system service. These rpc’s can be used to start, cancel and check status of reboot activities._
+
+
+## Requirements
+
+_NA_
+
+
+## Architecture Design
+
+The proposal is to create a Reboot Backend (BE) daemon in a new “Framework” container to handle a reboot request and forward it to the platform via DBUS.
+
+This daemon provides a location to implement application level reboot functionality before forwarding a reboot request to the platform (in the spirit of “[Docker to Host Communications HLD](https://github.com/sonic-net/SONiC/blob/master/doc/mgmt/Docker%20to%20Host%20communication.md)”.
+
+The reboot request contains a delay parameter, method and subcomponent fields. For subcomponents we can have support for:
```

Comment:
How do we use subcomponent to specify optic, container and etc?

Reply:
That would need to be designed: for now I'm treating empty subcomponent as a system wide switch reboot.

Being explicit: "we can have support". I'm not at this time proposing/designing support for optic or container reboot in this HLD.